ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -486,30 +486,46 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
 
     s3 = (S3_STORAGE *)storageSource;
 
-    /* We need to have a bucket to setup the file system. */
-    if (bucketName == NULL || strlen(bucketName) == 0) {
+    /*
+     * We need to have a bucket to setup the file system. The bucket is expected to be a name and a
+     * region, separated by a semi-colon. eg: 'abcd;ap-southeast-2'.
```

Comment:
This seems fine to me. I have a few questions/comments about it:
1. I don't know anything about the conventions around S3 buckets and regions to know: Would it make more sense to have it region;bucket instead? It seems region is the broader piece and bucket is finer-grained. Anyone have opinions?
2. Is a semi-colon the typical delimiter? I originally mis-read that as a colon because that is what I seemed to expect. Again, I don't know conventions. Are there conventions? If so, we should follow them.
3. Is this expectation documented anywhere? It needs to be somewhere in the documentation to make sure callers can find this information.

Reply:
I am not aware of the conventions either. I tried to find them, but most documents are around a legal alphabet set for the bucket name. ';' will work for now. I will file a ticket to talk to other (cloud) teams that know better. Also, there is a documentation ticket WT-8135, and I will file another documentation ticket to address the information required to build and use S3 extension.

Reply:
I created WT-8907 to both look for standards around bucket naming. I will probably talk to cloud team for that. The ticket will also update documentation.